### PR TITLE
Install packages using setup.py instead of debian packages

### PR DESCRIPTION
**Pull request by JamesGibo**
Using Anible to provision instead of shell script

### CODE REVIEW

File: vagrant/provisioning/roles/nmos-joint-ri-setup/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# description: Provision NMOS Joint RI VM
+
+- name: Install all required packages for nmos-joint-ri-setup
+  apt: pkg={{ packages }} state=present
+
+- name: Install Apache2 modules
+  apache2_module:
+    state: present
+    name: "{{ item }}"
+  with_items:
```
Comment (andrewbonney) [inline vagrant/provisioning/roles/nmos-joint-ri-setup/tasks/main.yml]: Perhaps the Apache modules could be added to the defaults?

**Review by andrewbonney — COMMENTED** [review]
Looks good. Just one specific comment as noted

File: vagrant/provisioning/roles/nmos-joint-ri-setup/tasks/main.yml (hunk 1 above)
Comment (JamesGibo) [reply vagrant/provisioning/roles/nmos-joint-ri-setup/tasks/main.yml]: Done

**Comment by JamesGibo** [comment]
All VMs have been switched to use pip to install packages
Also moved from script to Ansible for provisioning

File: vagrant/Vagrantfile [hunk 2]
```diff
@@ -86,8 +80,83 @@ Vagrant.configure("2") do |config|
...
+
+    auth_packages_to_install = packages_to_install + [{
+      "package_name" => "nmos-auth",
+      "repo_url" => "https://github.com/bbc/nmos-auth-server.git",
+      "branch" => repo_branches[7],
+      "service_file" => "python-nmos-auth.service",
+      "reverse_proxy_file" => "ips-api-auth.conf"
+    }]
+
+    node_packages_to_install = packages_to_install + [{
```
Comment (andrewbonney) [inline vagrant/Vagrantfile:136]: Does this not require nmoscommon/reverseproxy and mdnsbridge? Or are they installed via other means?

Auth server probably needs the first two as well.

**Review by andrewbonney — COMMENTED** [review]
Looks good. I suggest we hold off merging until next week!

File: vagrant/Vagrantfile (hunk 2 above)
Comment (JamesGibo) [reply vagrant/Vagrantfile:136]: There is a base list of packages that all VMs use which is concatenated with the VM specific one
`= packages_to_install + [{`

Comment (andrewbonney) [reply vagrant/Vagrantfile:136]: I see, didn't look closely enough!

**Comment by andrewbonney** [comment]
Looks like 'etcd' may be missing from the regquery VM. The tests work though so I need to look into why that is...

**Comment by JamesGibo** [comment]
> Looks like 'etcd' may be missing from the regquery VM. The tests work though so I need to look into why that is...

Added as a required debian dependency of the registration package

**Comment by JamesGibo** [comment]
Disabled dns discover and specified default interface

**Review by andrewbonney — APPROVED** [review]

**Pull request merged**